app/src/main/java/com/lagradost/cloudstream3/ui/account/AccountViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package com.lagradost.cloudstream3.ui.account
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.lagradost.cloudstream3.AcraApplication.Companion.context
+import com.lagradost.cloudstream3.ui.account.AccountHelper.showPinInputDialog
+import com.lagradost.cloudstream3.utils.DataStoreHelper
+import com.lagradost.cloudstream3.utils.DataStoreHelper.getAccounts
+import com.lagradost.cloudstream3.utils.DataStoreHelper.setAccount
+
+class AccountViewModel : ViewModel() {
+    private val _accounts: MutableLiveData<List<DataStoreHelper.Account>> = MutableLiveData(
+        context?.let { getAccounts(it) } ?: DataStoreHelper.accounts.toList()
+    )
+    val accounts: LiveData<List<DataStoreHelper.Account>> = _accounts
+
+    private val _isEditing = MutableLiveData(false)
+    val isEditing: LiveData<Boolean> = _isEditing
+
+    private val _isAllowedLogin = MutableLiveData(false)
+    val isAllowedLogin: LiveData<Boolean> = _isAllowedLogin
+
+    private val _selectedKeyIndex = MutableLiveData(DataStoreHelper.selectedKeyIndex)
+    val selectedKeyIndex: LiveData<Int> = _selectedKeyIndex
+
+    fun setIsEditing(value: Boolean) {
+        _isEditing.postValue(value)
+    }
+
+    fun toggleIsEditing() {
+        _isEditing.postValue(!(_isEditing.value ?: false))
+    }
+
+    fun updateAccounts(context: Context) {
```

Comment:
This should never be used, the viewmodel should instead handle all the logic like
```kotlin
import com.lagradost.cloudstream3.AcraApplication.Companion.context

fun deleteAccount(account : Account) {
	removeKeys(account.keyIndex.toString())
	val currentAccounts = DataStoreHelper.accounts.toMutableList()
	currentAccounts.removeIf { it.keyIndex == account.keyIndex }
	DataStoreHelper.accounts = currentAccounts.toTypedArray()
	
	if (account.keyIndex == DataStoreHelper.selectedKeyIndex) {
		setAccount(getDefaultAccount(context))
	}
        _accounts.postValue(getAccounts(context))
        _selectedKeyIndex.postValue(DataStoreHelper.selectedKeyIndex)
}
fun createAccount(account : Account) = handleAccountUpdate(account, context)
fun editAccount(account : Account) = handleAccountUpdate(account, context)
```
not via updateAccounts. Right now showAccountEditDialog is in charge of deleting the account and that makes no sense. The showAccountEditDialog should instead get the viewmodel and call the functions on it